src/nodes/hypertable_modify.c

```diff
@@ -745,7 +981,82 @@ ExecModifyTable(PlanState *pstate)
 	context.mtstate = node;
 	context.epqstate = &node->mt_epqstate;
 	context.estate = estate;
+	/*
+	 * For UPDATE/DELETE on compressed hypertable, decompress chunks and
+	 * move rows to uncompressed chunks.
+	 */
+	if ((operation == CMD_DELETE || operation == CMD_UPDATE) && !ht_state->comp_chunks_processed)
+	{
+		collect_chunks_from_scan(pstate, ht_state);
+		ListCell *ch;
+		ListCell *cf;
+		int pos = 0;
+		foreach (ch, ht_state->chunks)
+		{
+			bool chunk_status_changed = false;
+			PredicatesContext *cx = NULL;
+			List *scankeys = NIL;
+			List *null_columns = NIL;
+			List *is_null_checks = NIL;
 
+			if (ts_cm_functions->decompress_batches_for_update_delete)
+			{
+				if (ht_state->predicate_context)
+				{
+					cx = list_nth(ht_state->predicate_context, pos);
+					scankeys = cx->scankeys;
+					null_columns = cx->null_columns;
+					is_null_checks = cx->is_null_checks;
+					pos++;
+				}
+				if (ts_cm_functions->decompress_batches_for_update_delete(lfirst(ch),
+																		  scankeys,
+																		  null_columns,
+																		  is_null_checks,
+																		  &chunk_status_changed))
+				{
+					/*
+					 * tuples from compressed chunk has been decompressed and moved
+					 * to staging area, thus mark this chunk as partially compressed
+					 */
+					if (chunk_status_changed == true)
+						ts_chunk_set_partial(lfirst(ch));
+					/* free all chunks context */
+					if (scankeys)
+					{
+						foreach (cf, scankeys)
+							pfree((ScanKeyData *) lfirst(cf));
+					}
+					if (null_columns)
+					{
+						foreach (cf, null_columns)
+							pfree((Bitmapset *) linitial(null_columns));
+					}
+					if (cx)
+						pfree(cx);
+				}
+			}
+			ht_state->comp_chunks_processed = true;
+		}
+		if (ht_state->comp_chunks_processed)
+		{
+			/*
+			 * save snapshot set during ExecutorStart(), since this is the same
+			 * snapshot used to SeqScan of uncompressed chunks
+			 */
+			ht_state->snapshot = estate->es_snapshot;
+			/* use current transaction snapshot */
+			estate->es_snapshot = GetTransactionSnapshot();
```

Comment:
Why is this required, isn't `CommandCounterIncrement()` enough? I think Postgres almost never calls `GetTransactionSnapshot()` inside the executor code.

Reply:
If i dont have this code then, later when SeqScan nodes tries to fetch rows from uncompressed chunk, it will not find any rows. Sven mentioned that transaction snapshot should be same for batch filtering and Scan node.

For more details please refer: 
https://iobeam.slack.com/archives/C04QFNN1H6Y/p1677483665605009

Reply:
The slack link doesn't work for me, probably because I wasn't in that conversation at that point. Anyway, let's add a more detailed explanation to the code comment.

Reply:
Oops, sorry, added you to the group, i hope now you can access the chat.